### PR TITLE
Add ability to delete collection and workbook in detail page

### PR DESCRIPTION
**Pull request by Sergey-weber**

### CODE REVIEW

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx [hunk 1]
```diff
@@ -131,16 +134,35 @@ export const CollectionActions = React.memo<Props>(
...
+
+        if (collection && collection.permissions?.delete) {
+            otherActions.push({
+                text: <DropdownAction icon={TrashBin} text={i18n('action_delete')} />,
+                action: onDeleteClick,
+                theme: 'danger',
+            });
+        }
+
+        dropdownActions.push([...otherActions]);
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:156]:
```ts
if (otherActions.length > 0) {
    dropdownActions.push([...otherActions]);
}
```

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx [hunk 2]
```diff
@@ -118,16 +119,46 @@ export const WorkbookActions: React.FC<Props> = ({workbook, refreshWorkbookInfo}
...
+            action: () => {
+                dispatch(
+                    openDialog({
+                        id: DIALOG_DELETE_WORKBOOK,
+                        props: {
+                            open: true,
+                            workbookId: workbook.workbookId,
+                            workbookTitle: workbook.title,
+                            onSuccessApply: (id) => {
+                                dispatch(deleteWorkbookInItems(id));
```
Comment (imsitnikov) [inline src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx:141]: This is also not needed

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx [hunk 3]
```diff
@@ -91,12 +97,7 @@ export const WorkbookActions: React.FC<Props> = ({workbook, refreshWorkbookInfo}
...
-            text: (
-                <React.Fragment>
-                    <Icon data={ArrowRight} className={classNameIconAction} />
-                    {i18n('action_move')}
-                </React.Fragment>
-            ),
+            text: <DropdownAction icon={ArrowRight} text={i18n('action_move')} />,
         });
 
         if (workbook.permissions.copy) {
```
Comment (imsitnikov) [inline src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx:103]: This is an old change, but why did we put two nested `if` here?
```ts
if (workbook.permissions.move) {
    if (workbook.permissions.copy) {
        ...
    }
}
```

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx (hunk 3 above)
Comment (Sergey-weber) [reply src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx]: probably here unnesessary put two nested if, fixed it

File: src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx [hunk 4]
```diff
@@ -174,6 +175,34 @@ export const useLayout = ({
...
+                                if (curCollectionId && collection) {
+                                    dispatch(
+                                        openDialog({
+                                            id: DIALOG_DELETE_COLLECTION,
+                                            props: {
+                                                open: true,
+                                                collectionId: curCollectionId,
+                                                collectionTitle: collection.title,
+                                                onSuccessApply: (id) => {
+                                                    dispatch(deleteCollectionInItems(id));
```
[the remark this answers is not in the document]
Comment (Sergey-weber) [reply src/ui/units/collections/components/CollectionPage/hooks/useLayout.tsx]: yes i agree, fixed it, thanks

**Review by imsitnikov — APPROVED** [review]

**Pull request merged**